lib/Dancer2/Core/Role/SessionFactory.pm

```diff
@@ -163,7 +164,7 @@ sub retrieve {
     my %args = ( id => $id, );
 
     $args{data} = $data
-      if $data and ref $data eq 'HASH';
+      if $data and ref $data and reftype $data eq 'HASH';
```

Comment:
@yanick Is there any chance of $data being a scalar? My version avoids the following warning:
```
$ perl -MScalar::Util=reftype -we'my $data = 1; print "OK" if reftype $data eq "HASH"'
Use of uninitialized value in string eq at -e line 1.

```